### DISCUSSION TITLE
What is the best way to rearrange the lattice vectors of a structure object?

### DISCUSSION THREAD

**Discussion by mm04926412** [materialsproject/pymatgen#2638] [Q&A] [answered]
What methods (if any) are there to rearrange the lattice parameters of a structure object (swap a and b over etc.) without breaking it? I would assume that directly changing a b and c in the structures lattice property would result in the structure object becoming invalid more than anything? I'm doing some work that isnt commutative with respect to the lattice parameters so I wanted to be able to try every combination of a,b and c and return the best result.

**Comment by shyuep** [comment, marked as answer]
You can apply a transformation matrix to the lattice parameters. See Structure.apply_transformation.

**Comment by mm04926412** [comment]
ah thanks thats pretty obvious a solution in retrospect, thank you for answering both my questions!
